Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/detail/AgendaDetailFragment.kt

```diff
@@ -0,0 +1,182 @@
+import android.support.design.widget.Snackbar
+import android.support.v4.content.ContextCompat
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.RelativeLayout
+import com.bumptech.glide.Glide
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.ValueEventListener
+import com.google.gson.Gson
+import com.mentalmachines.droidcon_boston.R
+import com.mentalmachines.droidcon_boston.R.string
+import com.mentalmachines.droidcon_boston.data.FirebaseDatabase.ScheduleEventDetail
+import com.mentalmachines.droidcon_boston.data.Schedule
+import com.mentalmachines.droidcon_boston.data.Schedule.ScheduleDetail
+import com.mentalmachines.droidcon_boston.data.Schedule.ScheduleRow
+import com.mentalmachines.droidcon_boston.data.UserAgendaRepo
+import com.mentalmachines.droidcon_boston.firebase.FirebaseHelper
+import com.mentalmachines.droidcon_boston.utils.StringUtils
+import com.mentalmachines.droidcon_boston.views.agenda.CircleTransform
+import kotlinx.android.synthetic.main.agenda_detail_fragment.agendaDetailView
+import kotlinx.android.synthetic.main.agenda_detail_fragment.fab_agenda_detail_bookmark
+import kotlinx.android.synthetic.main.agenda_detail_fragment.tv_agenda_detail_description
+import kotlinx.android.synthetic.main.agenda_detail_fragment.tv_agenda_detail_room
+import kotlinx.android.synthetic.main.agenda_detail_fragment.tv_agenda_detail_speaker_name
+import kotlinx.android.synthetic.main.agenda_detail_fragment.tv_agenda_detail_speaker_title
+import kotlinx.android.synthetic.main.agenda_detail_fragment.tv_agenda_detail_time
+import kotlinx.android.synthetic.main.agenda_detail_fragment.tv_agenda_detail_title
+
+
+class AgendaDetailFragment : Fragment() {
+
+    private lateinit var scheduleDetail: ScheduleDetail
+
+    private val gson: Gson = Gson()
+
+    private val firebaseHelper = FirebaseHelper.instance
+
+    private val userAgendaRepo: UserAgendaRepo
+        get() = UserAgendaRepo.getInstance(fab_agenda_detail_bookmark.context)
+
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        super.onCreateView(inflater, container, savedInstanceState)
+
+        return inflater.inflate(R.layout.agenda_detail_fragment, container, false)
+    }
+
+    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val itemData = gson.fromJson(arguments.getString(Schedule.SCHEDULE_ITEM_ROW), ScheduleRow::class.java)
+        fetchDataFromFirebase(itemData)
+        populateView(itemData)
+    }
+
+    private fun populateView(itemData: ScheduleRow) {
+        tv_agenda_detail_title.text = itemData.talkTitle
+        tv_agenda_detail_room.text = "at " + itemData.room
```

Comment:
Can probably do string interpolation

Reply:
Was doing that earlier, AS suggested i convert to concatenation

Reply:
Weird..wonder why AS wanted you to do that?

Reply:
![screen shot 2018-03-03 at 6 35 47 am](https://user-images.githubusercontent.com/2096087/36928595-307a8fec-1ead-11e8-8544-083ddac6cd60.png)

The quick fix after converting to string interpolation version.

Reply:
Nevermind I fixed it.